hpx/runtime/threads/policies/queue_holder_numa.hpp

```diff
@@ -0,0 +1,303 @@
+                }
+                // if stealing disabled, do not check other queues
+                if (!allow_stealing)
+                    return false;
+            }
+            return false;
+        }
+
+        // ----------------------------------------------------------------
+        inline std::size_t get_new_tasks_queue_length() const
+        {
+            std::size_t len = 0;
+            for (auto& q : queues_)
+                len += q->new_tasks_count_;
+            return len;
+        }
+
+        // ----------------------------------------------------------------
+        inline std::int64_t get_thread_count(thread_state_enum state = unknown,
+            thread_priority priority = thread_priority_default) const
+        {
+            std::size_t len = 0;
+            for (auto& q : queues_)
+                len += q->get_thread_count(state, priority);
+            return len;
+        }
+
+        // ----------------------------------------------------------------
+        void abort_all_suspended_threads()
+        {
+            for (auto& q : queues_)
+                q->abort_all_suspended_threads();
+        }
+
+        // ----------------------------------------------------------------
+        bool enumerate_threads(
+            util::function_nonser<bool(thread_id_type)> const& f,
+            thread_state_enum state) const
+        {
+            bool result = true;
+            for (auto& q : queues_)
+                result = q->enumerate_threads(f, state) && result;
+            return result;
+        }
+
+        // ----------------------------------------------------------------
+        // ----------------------------------------------------------------
+        // ----------------------------------------------------------------
+        std::uint16_t num_queues_;
+        std::uint16_t domain_;
+        std::vector<ThreadQueue*> queues_;
+
+    public:
+        //        // ------------------------------------------------------------
+        //        // This returns the current length of the pending queue
+        //        std::int64_t get_pending_queue_length() const
+        //        {
+        //            return work_items_count_;
+        //        }
+
+        //        // This returns the current length of the staged queue
+        //        std::int64_t get_staged_queue_length(
+        //            std::memory_order order = std::memory_order_seq_cst) const
+        //        {
+        //            return new_tasks_count_.load(order);
+        //        }
+
+        void increment_num_pending_misses(std::size_t num = 1) {}
+        void increment_num_pending_accesses(std::size_t num = 1) {}
+        void increment_num_stolen_from_pending(std::size_t num = 1) {}
+        void increment_num_stolen_from_staged(std::size_t num = 1) {}
+        void increment_num_stolen_to_pending(std::size_t num = 1) {}
+        void increment_num_stolen_to_staged(std::size_t num = 1) {}
```

Comment:
Will those counters be available for the new scheduler?

Reply:
I've not yet needed them but I am conscious of the fact that they're missing. I actually find the total number of tasks executed by each thread to be a useful indicator. Since the scheduler supports `thread_parent` or `round_robin` injection of tasks we expect the number stolen to be large when many tasks are created by a single thread using `thread_parent` (example - the futures_overhad test) - when thread parent mode is used, it might be the case that all task are created by one thread, so stolen numbers will always be huge. with round robin creation, one expects less. Ultimately, the idle rate and number of tasks executed by each thread gives a good estimate of how well the tasks are being executed.
I didn't provide a counter for the total executed, though I did have an internal counter on one branch that I removed during cleaning. Do we have an existing counter for tasks executed that I could use for it?

Reply:
NB. Would a histogram counter be possible for the tasks executed per core?